presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/JdbcResourceGroupConfigSource.java

```diff
@@ -0,0 +1,449 @@
+            "soft_memory_limit VARCHAR(128) NOT NULL,\n" +
+            "max_queued INT NOT NULL,\n" +
+            "max_running INT NOT NULL,\n" +
+            "scheduling_policy VARCHAR(128),\n" +
+            "scheduling_weight INT,\n" +
+            "parent_id INT,\n" +
+            "jmx_export BOOLEAN\n" +
+            ");\n" +
+            "\n" +
+            "CREATE TABLE %s (\n" +
+            "group_name VARCHAR(512) NOT NULL,\n" +
+            "user_regex VARCHAR(512),\n" +
+            "source_regex VARCHAR(512)\n" +
+            ");", RESOURCE_GROUP_TABLE, SELECTOR_TABLE);
+
+    private final Provider<Connection> connectionProvider;
+    private final ManagerSpec managerSpec;
+
+    @Inject
+    public JdbcResourceGroupConfigSource(Provider<Connection> connectionProvider)
+    {
+        //TODO: use airlift.dbpool
+        this.connectionProvider = requireNonNull(connectionProvider, "Connection provider is null");
+        checkOrCreateMetadata();
+        ResourceGroupDbRecords resourceGroupDbRecords = new ResourceGroupDbRecords(getResourceGroups());
+        managerSpec = new ManagerSpec(resourceGroupDbRecords.getRootGroups(), getSelectors());
+    }
+
+    @Override
+    public ManagerSpec get()
+    {
+        return managerSpec;
+    }
+
+    private List<ResourceGroupDbRecord> getResourceGroups()
+    {
+        Statement statement = null;
+        ResultSet result = null;
+        ImmutableList.Builder<ResourceGroupDbRecord> records = ImmutableList.builder();
+        try {
+            Connection connection = connectionProvider.get();
+            statement = connection.createStatement();
+            result = statement.executeQuery(RESOURCE_GROUP_QUERY);
+            while (result.next()) {
+                int resourceGroupRecordId = result.getInt(1);
+                String name = result.getString(2);
+                String softMemoryLimit = result.getString(3);
+                int maxQueued = result.getInt(4);
+                int maxRunning = result.getInt(5);
+                Optional<String> schedulingPolicy = Optional.of(result.getString(6));
+                Optional<Integer> schedulingWeight = Optional.of(result.getInt(7));
+                Optional<Integer> parentResourceGroupRecordId = Optional.of(result.getInt(8));
+                Optional<Boolean> jmxExport = Optional.of(result.getBoolean(9));
+                records.add(new ResourceGroupDbRecord(
+                        resourceGroupRecordId,
+                        name,
+                        softMemoryLimit,
+                        maxQueued,
+                        maxRunning,
+                        schedulingPolicy,
+                        schedulingWeight,
+                        parentResourceGroupRecordId,
+                        jmxExport));
+            }
+        }
+        catch (SQLException e) {
+            throw new RuntimeException(format("Error retrieving resource group metadata: %s", e));
+        }
+        finally {
```

Comment:
This all goes away with JDBI, but as of Java 7, you can use try-with-resources and everything will get closed properly:
```java
try (Connection connection = connectionProvider.get(); 
        Statement statement = connection.createStatement(); 
        ResultSet rs = statement.executeQuery(...)) {
    ....
}